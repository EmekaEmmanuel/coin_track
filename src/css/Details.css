* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Montserrat', sans-serif;
  letter-spacing: 0.2px;
}

.app_container {
  margin: auto;
  background-color: #111315;
}

.details_section1 {
  padding: 100px 0;
  width: 50%;
  margin: auto;
}

.details_font1 {
  font-size: 11px;
  color: #efefef9a;
}

.details_font2 {
  font-size: 50px;
  color: #efefef;
  padding: 20px 0;
}

.details_font3 {
  font-size: 11px;
  color: #efefef9a;
}

.details_section2 {
  width: 90%;
  margin: auto;
  padding: 30px 30px;
  background-color: #020a11;
  border-radius: 15px;
}

.details_section2_figure img {
  width: 300px;
  object-position: center;
  object-fit: cover;
}

.details_section_listbox {
  list-style: none;
  color: #efefef;
  width: 60%;
  margin: auto;
}

.details_section_listbox li {
  display: flex;
  justify-content: space-between;
  padding: 5px 0;
  color: #efefef9a;
}

.details_button_box {
  padding: 20px 0 0 0;
  display: flex;
  justify-content: center;
  gap: 20px;
  text-align: center;
}

.details_chart_btn {
  padding: 8px 15px;
  background-color: #0077fe;
  color: #efefef;
  border: none;
  border-radius: 10px;
}

.details_chart_btn a {
  color: #efefef;
}
