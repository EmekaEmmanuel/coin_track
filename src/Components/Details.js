import React from 'react';
import { Link, useParams } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import '../css/Details.css';

function Details(props) {
  const { myCoins } = props;
  const { id } = useParams();

  const coin = (myCoins.filter((coin) => coin.id === id))[0];

  return (
    <div className="app_container">
      <main className="main_tag">

        <section className="details_section1">
          <p className="details_font1">LIVE MARKET CHANGES</p>
          <h1 className="details_font2">
            Get real time details on
            {' '}
            {coin.name}
          </h1>
          <p className="details_font3">Your favorite place to see what&#39;s new.</p>
        </section>

        {coin
          && (
            <section className="details_section2" key={coin.id}>
              <figure className="details_section2_figure">
                <img src={coin.icon} alt="icon" />
              </figure>

              <ul className="details_section_listbox">
                <li>
                  <p>name</p>
                  <p>{coin.name}</p>
                </li>
                <li>
                  <p>symbol</p>
                  <p>{coin.symbol}</p>
                </li>
                <li>
                  <p>rank</p>
                  <p>{coin.rank}</p>
                </li>
                <li>
                  <p>price</p>
                  <p>{(coin.price).toFixed(2)}</p>
                </li>
                <li>
                  <p>priceBtc</p>
                  <p>{(coin.priceBtc).toFixed(2)}</p>
                </li>
                <li>
                  <p>volume</p>
                  <p>{(coin.volume).toFixed(2)}</p>
                </li>
                <li>
                  <p>marketCap</p>
                  <p>{(coin.marketCap).toFixed(2)}</p>
                </li>
                <li>
                  <p>availableSupply</p>
                  <p>{coin.availableSupply}</p>
                </li>
                <li>
                  <p>totalSupply</p>
                  <p>{coin.totalSupply}</p>
                </li>
                <li>
                  <p>priceChange1h</p>
                  <p>{coin.priceChange1h}</p>
                </li>
                <li>
                  <p>priceChange1d</p>
                  <p>{coin.priceChange1d}</p>
                </li>
                <li>
                  <p>priceChange1w</p>
                  <p>{coin.priceChange1w}</p>
                </li>
                <li>
                  <p>websiteUrl</p>
                  <p>{coin.websiteUrl}</p>
                </li>
                <li>
                  <p>twitterUrl</p>
                  <p>{coin.twitterUrl}</p>
                </li>
              </ul>

              <div className="details_button_box">
                <button className="details_chart_btn" type="button">
                  <Link to={`/chart/${coin.id}`} className="nav_font2 td_none">View Live chart </Link>
                </button>

                <button className="details_chart_btn" type="button">
                  <Link to={`/market/${coin.id}`} className="nav_font2 td_none">View Markets </Link>
                </button>
              </div>
            </section>
          )}
      </main>
    </div>
  );
}

Details.propTypes = {
  myCoins: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.object)).isRequired,
};

export default Details;
