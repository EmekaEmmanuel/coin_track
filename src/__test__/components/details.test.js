import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import Details from '../../Components/Details';

it('Details component renders correctly', () => {
  const myCoins = store.getState().coin.coins;
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Details myCoins={myCoins} />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
