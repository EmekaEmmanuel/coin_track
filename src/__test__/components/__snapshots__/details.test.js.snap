// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component renders correctly 1`] = `
<div
  className="app_container"
>
  <main
    className="main_tag"
  >
    <section
      className="details_section1"
    >
      <p
        className="details_font1"
      >
        LIVE MARKET CHANGES
      </p>
      <h1
        className="details_font2"
      >
        Get real time
         
         
        details
      </h1>
      <p
        className="details_font3"
      >
        Your favorite place to see what's new.
      </p>
    </section>
  </main>
</div>
`;
